src/bors/handlers/review.rs

```diff
@@ -598,4 +667,164 @@ approve = ["+approved"]
+            Ok(tester)
+        })
+        .await;
+    }
+
+    #[sqlx::test]
+    async fn delegatee_can_approve(pool: sqlx::PgPool) {
+        BorsBuilder::new(pool)
+            .world(create_world_with_approve_config())
+            .run_test(|mut tester| async {
+                let pr_author = User::default();
+                tester.post_comment("@bors delegate+").await?;
+                tester.expect_comments(1).await;
+
+                tester
+                    .post_comment(Comment::from("@bors r+").with_author(pr_author.clone()))
+                    .await?;
+                tester.expect_comments(1).await;
+
+                check_pr_approved_by(&tester, default_pr_number().into(), &pr_author.name).await;
+                Ok(tester)
+            })
+            .await;
+    }
+
+    #[sqlx::test]
+    async fn delegatee_can_try(pool: sqlx::PgPool) {
+        run_test(pool, |mut tester| async {
+            let pr_author = User::default();
+
+            tester.post_comment("@bors delegate+").await?;
+            tester.expect_comments(1).await;
+
+            tester
+                .post_comment(Comment::from("@bors try").with_author(pr_author.clone()))
+                .await?;
+            tester.expect_comments(1).await;
+
+            Ok(tester)
+        })
+        .await;
+    }
+
+    #[sqlx::test]
+    async fn delegate_insufficient_permission(pool: sqlx::PgPool) {
+        let world = World::default();
+        world.default_repo().lock().permissions = Permissions::default();
+
+        BorsBuilder::new(pool)
+            .world(world)
+            .run_test(|mut tester| async {
+                tester.post_comment("@bors delegate+").await?;
+                assert_eq!(
+                    tester.get_comment().await?,
+                    "@default-user: :key: Insufficient privileges: not in review users"
+                );
+
+                let pr = tester.get_default_pr().await?;
+                assert!(!pr.delegated);
+                Ok(tester)
+            })
+            .await;
+    }
+
+    #[sqlx::test]
+    async fn undelegate_by_reviewer(pool: sqlx::PgPool) {
+        run_test(pool, |mut tester| async {
+            tester.post_comment("@bors delegate+").await?;
+            tester.expect_comments(1).await;
+
+            let pr = tester.get_default_pr().await?;
+            assert!(pr.delegated);
+
+            tester.post_comment("@bors delegate-").await?;
+            tester.expect_comments(1).await;
+
+            let pr = tester.get_default_pr().await?;
+            assert!(!pr.delegated);
+            Ok(tester)
+        })
+        .await;
+    }
+
+    #[sqlx::test]
+    async fn undelegate_by_delegatee(pool: sqlx::PgPool) {
+        run_test(pool, |mut tester| async {
+            tester.post_comment("@bors delegate+").await?;
+            tester.expect_comments(1).await;
+
+            tester
+                .post_comment(Comment::from("@bors delegate-").with_author(User::default()))
```

Comment:
```suggestion
                .post_comment(Comment::from("@bors delegate-").with_author(User::default()))
```
Again here, `User::default()` is the author of the PR, so this whole test deals just with a single user.

Reply:
I'm a doughnut, not too sure what I was thinking in some parts of the code 😅. Sorry for the silly errors, they should be fixed.